### PR TITLE
Add Scan Report permissions

### PR DESCRIPTION
**Pull request by AndyRae**
# Changes

Adds a permissions filter to Scan Reports, tables, fields, and values views. 
These permissions are based on the existing in `permissions.py`

Related #776 

# Checks

**Important:** please complete these **before** merging.
- [x] Run migrations, if any.
- [x] Update `changelog.md`, including migration instructions if any.
- [x] Run unit tests.

### CODE REVIEW

**Review by dcl10 — APPROVED** [review]

File: app/api/api/views.py [hunk 1]
```diff
@@ -1005,53 +1006,6 @@ class ScanReportConceptFilterViewSet(viewsets.ModelViewSet):
     }
 
 
-class ScanReportActiveConceptFilterViewSet(viewsets.ModelViewSet):
```
Comment (spco) [inline app/api/api/views.py:1008]: Has this been superceded already? This is/was the function used by the auto-mapping functions to look across all Active SRs for reusable mappings.

File: app/api/api/views.py [hunk 2]
```diff
@@ -1266,18 +1220,6 @@ def get_serializer_class(self):
         return super().get_serializer_class()
 
 
-class ScanReportFilterViewSet(viewsets.ModelViewSet):
```
Comment (spco) [inline app/api/api/views.py:1269]: Have these all been superceded elsewhere?

File: app/api/api/filters.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        dataset_visibility = f"{relationship}parent_dataset__visibility"
+        scan_report_visibility = f"{relationship}visibility"
+        scan_report_viewers = f"{relationship}viewers"
+        scan_report_editors = f"{relationship}editors"
+        scan_report_author = f"{relationship}author"
+        dataset_viewers = f"{relationship}parent_dataset__viewers"
+        dataset_editors = f"{relationship}parent_dataset__editors"
+        dataset_admins = f"{relationship}parent_dataset__admins"
+
+        return queryset.filter(
```
Comment (spco) [inline app/api/api/filters.py:44]: I haven't looked at the code for too long to be sure this isn't implemented elsewhere: but I think here we also need to implement Project-level filtering. Users should never be able to see a SR that isn't in one of the Projects they are a member of, even if that SR and its Dataset are set as "PUBLIC".

The only exception is AZ_FUNCTION_USER who can see everything to allow mapping reuse.

File: app/api/api/views.py (hunk 1 above)
Comment (AndyRae) [reply app/api/api/views.py:1008]: Yes the mapping reuse is completely database driven now - no API is used.

File: app/api/api/views.py (hunk 2 above)
Comment (AndyRae) [reply app/api/api/views.py:1269]: Yes they are no longer in use.

File: app/api/api/filters.py (hunk 3 above)
Comment (AndyRae) [reply app/api/api/filters.py:44]: Thanks - hadn't thought of the project level, I will add this.

Comment (dcl10) [reply app/api/api/filters.py:45]: good spot @spco

**Review by dcl10 — APPROVED** [review]

**Review by AndrewThien — APPROVED** [review]

File: app/api/api/filters.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+        dataset_visibility = f"{relationship}parent_dataset__visibility"
+        scan_report_visibility = f"{relationship}visibility"
+        scan_report_viewers = f"{relationship}viewers"
+        scan_report_editors = f"{relationship}editors"
+        scan_report_author = f"{relationship}author"
+        dataset_viewers = f"{relationship}parent_dataset__viewers"
+        dataset_editors = f"{relationship}parent_dataset__editors"
+        dataset_admins = f"{relationship}parent_dataset__admins"
+        project_members = f"{relationship}parent_dataset__project__members"
```
Comment (spco) [inline app/api/api/filters.py:43]: I'm going to assume that you've checked this works, remembering that datasets to projects is many-to-many.

**Review by spco — APPROVED** [review]

File: app/api/api/filters.py (hunk 4 above)
Comment (AndyRae) [reply app/api/api/filters.py:43]: Yep it works, and with multiple users involved (as in, multiple users being in the lists) - I was expecting to use `__in` (or however Django does it) but it wasn't necessary.

**Pull request merged**